crates/forge/tests/fixtures/zk/ScriptSetup.s.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {Script} from "forge-std/Script.sol";
+import {Greeter} from "../src/Greeter.sol";
+
+contract ScriptSetupNonce is Script {
+    function setUp() public {
+        uint256 initial_nonce = checkNonce(address(tx.origin));
+        // Perform transactions and deploy contracts in setup to increment nonce and verify broadcast nonce matches onchain
+        new Greeter();
+        new Greeter();
+        new Greeter();
+        new Greeter();
+        assert(checkNonce(address(tx.origin)) == initial_nonce);
+    }
+
+    function run() public {
+        uint256 initial_nonce = checkNonce(address(tx.origin));
+        assert(initial_nonce == vm.getNonce(address(tx.origin)));
+        Greeter notBroadcastGreeter = new Greeter();
+        notBroadcastGreeter.greeting("john");
+        assert(checkNonce(address(tx.origin)) == initial_nonce);
+        vm.startBroadcast();
+        Greeter greeter = new Greeter();
+        greeter.greeting("john");
+        NonceChecker checker = new NonceChecker();
+        checker.assertNonce(vm.getNonce(address(tx.origin)) + 1);
+        vm.stopBroadcast();
+    }
+
+    function checkNonce(address addr) public returns (uint256) {
+        vm.prank(address(this), address(this));
```

Comment:
Not necessary, but to make sure we don't accidentally pass `address(this)` to the function, we could use a random address here (dealing it some tokens for the call).
Alternatively, using `sload` cheatcode to load the storage slot directly but we don't know how to compute it
```suggestion
        // We prank here to avoid accidentally "polluting" the nonce of `addr` during the call
        // for example when `addr` is `tx.origin`
        vm.prank(address(this), address(this));
```